modules/user/routes.js

```diff
@@ -1,21 +1,28 @@
-import express from 'express';
-const router = express.Router()
-import createUserValidator from './validators/createUserValidator.js'
-import * as controller from './controller.js'
-import authenticationMiddleware from '../../middlewares/authentication.js'
+import express from "express";
+const router = express.Router();
+import createUserValidator from "./validators/createUserValidator.js";
+import validateLikes from "./validators/createLikeValidator.js";
+import * as controller from "./controller.js";
+import authenticationMiddleware from "../../middlewares/authentication.js";
 
-router.post('/signup', createUserValidator, controller.create)
+router.post("/signup", createUserValidator, controller.create);
 
-router.post('/login', controller.login)
+router.post("/login", controller.login);
 
-router.get('/', controller.find)
+router.get("/", controller.find);
+router.get("/posts", controller.readPosts);
+router.get("/:id", controller.findById);
+router.get("/posts/comments/:id", controller.readComments);
 
-router.get('/:id', controller.findById)
+router.use(authenticationMiddleware);
 
-router.use(authenticationMiddleware)
+router.post("/addPost", controller.addPost);
+router.post("/posts/:id", controller.addComment);
+router.post("/posts/like/:id",validateLikes, controller.addPostLike);
+router.post("/posts/comment/like/:id",validateLikes, controller.addCommentLike);
```

Comment:
```suggestion
```

you should add these apis in a separate router in a separate module to keep things organized